lib/ex_vault/utils.ex

```diff
@@ -4,4 +4,19 @@ defmodule ExVault.Utils do
   def uri_encode(val) when is_integer(val), do: to_string(val)
   def uri_encode(val) when is_atom(val), do: val |> to_string() |> uri_encode()
   def uri_encode(val), do: URI.encode(val, &URI.char_unreserved?/1)
+
+  def to_uri_path(secret_path) when is_list(secret_path),
```

Comment:
Please rename this function name and argument to general names here. It was private function only taking lists for secret path - but now this is in general-urpose Utils module.